libs/src/api/Track.ts

```diff
@@ -533,15 +559,30 @@ export class Track extends Base {
             trackInfo
 
           // Write metadata to chain
-          const multihashDecoded = Utils.decodeMultihash(metadataMultihash)
-          const { txReceipt, trackId } =
-            await this.contracts.TrackFactoryClient.addTrack(
+          let txReceipt: TransactionReceipt
+          let trackId: number
+          if (useEntityManager) {
+            trackId = Track.generateTrackId()
+            const response =
+              await this.contracts.EntityManagerClient!.manageEntity(
```

Comment:
just wondering why it's nullable here and if there is anything clever we can do about it, nbd tho

Reply:
Because we are adding a new contract, the entity manager addr might not be passed in and is nullable - https://github.com/AudiusProject/audius-protocol/blob/7c68b2a99e1942173664a26a97f2b9432cb60ca3/libs/src/services/dataContracts/AudiusContracts.ts#L48